### ISSUE TITLE
QC generation stage fails when no fragments are generated

### ISSUE THREAD

**Issue by j-wags** [openforcefield/bespoke-fit#88]
The following command fails when using a fresh install from `main` 

`openff-bespoke executor run --input BrCO.sdf --spec-file schema.json --n-qc-compute-workers 4`

`BrCO.sdf` and `schema.json` (and the `run.py` file that makes them but crashes in the process )can be found here: 
[inputs.tar.gz](https://github.com/openforcefield/bespoke-fit/files/7508126/inputs.tar.gz)


Env created by appending `xtb-python` to the test env, then using `conda env create -f devtools/conda-envs/test-env.yaml`. Full `conda list` is below.

This setup failed to run on @bergazin's macbook, my macbook, and a linux docker container on @bergazin's macbook. I also tried 
* replacing `--spec-file` with `--spec default`
* removing `--n-qc-compute-workers`
* running with AmberTools+RDKit instead of OE
* running from c087e0eccd43fc0a0039d808281c8f4ac9b8a207 instead 
* running from dffb29e5d81bbdf0e16482dde40e1baa62c1e65f instead
* Downgrading qcelemental and qcengine

Output:

<details>
<summary></summary>
<br>

```
(test) jeffreywagner@JW-MBP$ openff-bespoke executor run --input BrCO.sdf --spec-file schema.json --n-qc-compute-workers 4

─────────────────────────────────────────────────────────── OpenFF Bespoke ───────────────────────────────────────────────────────────

[✓] bespoke executor launched

1. preparing the bespoke workflow                                                                                                     
                                                                                                                                      
[✓] 1 molecule was found
[✓] fitting schema generated
                                                                                                                                      
2. submitting the workflow                                                                                                            
                                                                                                                                      
[✓] workflow submitted: id=1
                                                                                                                                      
3. running the fitting pipeline                                                                                                       
                                                                                                                                      
[✓] fragmentation successful
[x] qc-generation failed
                                                                                                                                      
 "Internal Server Error"                                                                                                              

worker: Warm shutdown (MainProcess)

worker: Warm shutdown (MainProcess)

worker: Warm shutdown (MainProcess)
```

The file `bespoke_executor/gateway.log` contains the following error:

```
(test) jeffreywagner@JW-MBP$ cat bespoke-executor/gateway.log 
[task id=b'coordinator:optimization:1'] transitioned from waiting -> running
[task id=b'coordinator:optimization:1'] fragmentation transitioned from waiting -> running
[task id=b'coordinator:optimization:1'] fragmentation transitioned from running -> success
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 373, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/middleware/cors.py", line 78, in __call__
    await self.app(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/fastapi/routing.py", line 219, in app
    raw_response = await run_endpoint_function(
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/fastapi/routing.py", line 154, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/starlette/concurrency.py", line 40, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/openff_bespokefit-27rc4904+28.g739f680.dirty-py3.9.egg/openff/bespokefit/executor/services/qcgenerator/app.py", line 70, in get_qc_results
    contents=[_retrieve_qc_result(qc_calc_id, results) for qc_calc_id in ids],
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/openff_bespokefit-27rc4904+28.g739f680.dirty-py3.9.egg/openff/bespokefit/executor/services/qcgenerator/app.py", line 70, in <listcomp>
    contents=[_retrieve_qc_result(qc_calc_id, results) for qc_calc_id in ids],
  File "/Users/jeffreywagner/miniconda3/envs/test/lib/python3.9/site-packages/openff_bespokefit-27rc4904+28.g739f680.dirty-py3.9.egg/openff/bespokefit/executor/services/qcgenerator/app.py", line 46, in _retrieve_qc_result
    "type": qc_calc_type.decode(),
AttributeError: 'NoneType' object has no attribute 'decode'
[task id=b'coordinator:optimization:1'] qc-generation transitioned from waiting -> errored
[task id=b'coordinator:optimization:1'] transitioned from running -> errored
```

</details>

Env info:

<details>
<summary></summary>
<br>

```
(test) jeffreywagner@JW-MBP$ conda list 
# packages in environment at /Users/jeffreywagner/miniconda3/envs/test:
#
# Name                    Version                   Build  Channel
abseil-cpp                20210324.2           he49afe7_0    conda-forge
amberlite                 16.0                     pypi_0    pypi
ambertools                21.11            py39hf80593e_0    conda-forge
amberutils                21.0                     pypi_0    pypi
amqp                      5.0.6              pyhd8ed1ab_0    conda-forge
anyio                     3.3.4            py39h6e9494a_0    conda-forge
appnope                   0.1.2            py39h6e9494a_2    conda-forge
argcomplete               1.12.3             pyhd8ed1ab_2    conda-forge
argon2-cffi               21.1.0           py39h89e85a6_2    conda-forge
arpack                    3.7.0                hefb7bc6_2    conda-forge
arrow-cpp                 6.0.0           py39h1bf2758_0_cpu    conda-forge
asgiref                   3.4.1              pyhd8ed1ab_0    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
aws-c-auth                0.6.4                hd0defc5_3    conda-forge
aws-c-cal                 0.5.12               h88d0821_2    conda-forge
aws-c-common              0.6.11               h0d85af4_0    conda-forge
aws-c-compression         0.2.14               h4dc997f_2    conda-forge
aws-c-event-stream        0.2.7               h0d10642_24    conda-forge
aws-c-http                0.6.6                h3059b7b_1    conda-forge
aws-c-io                  0.10.9               h42eeb19_3    conda-forge
aws-c-mqtt                0.7.8                h7ee32f5_2    conda-forge
aws-c-s3                  0.1.27               h14340f6_1    conda-forge
aws-checksums             0.1.12               h4dc997f_1    conda-forge
aws-crt-cpp               0.17.1               hf6c2175_3    conda-forge
aws-sdk-cpp               1.9.120              hb246b62_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
basis_set_exchange        0.8.13                     py_0    conda-forge
billiard                  3.6.4.0          py39h89e85a6_1    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.0               he49afe7_0    conda-forge
boost                     1.74.0           py39ha1f3e3e_4    conda-forge
boost-cpp                 1.74.0               hf3dc895_5    conda-forge
brotlipy                  0.7.0           py39h89e85a6_1003    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
c-ares                    1.18.1               h0d85af4_0    conda-forge
ca-certificates           2021.10.8            h033912b_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cachetools                4.2.4              pyhd8ed1ab_0    conda-forge
cairo                     1.16.0            he01c77b_1009    conda-forge
celery                    5.1.2              pyhd8ed1ab_0    conda-forge
certifi                   2021.10.8        py39h6e9494a_1    conda-forge
cffi                      1.15.0           py39he338e87_0    conda-forge
chardet                   4.0.0            py39h6e9494a_2    conda-forge
charset-normalizer        2.0.0              pyhd8ed1ab_0    conda-forge
chemper                   1.0.0              pyh9f0ad1d_0    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-didyoumean          0.0.3              pyh8c360ce_0    conda-forge
click-option-group        0.5.3              pyhd8ed1ab_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
click-repl                0.2.0              pyhd8ed1ab_0    conda-forge
codecov                   2.1.11             pyhd3deb0d_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
commonmark                0.9.1                      py_0    conda-forge
coverage                  6.1.1            py39h89e85a6_1    conda-forge
cryptography              35.0.0           py39h209aa08_2    conda-forge
curl                      7.79.1               hf45b732_1    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cython                    0.29.24          py39h9fcab8e_1    conda-forge
debugpy                   1.5.1            py39h9fcab8e_0    conda-forge
decorator                 5.1.0              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.3             py39hde42818_1002    conda-forge
fastapi                   0.67.0             pyhd8ed1ab_0    conda-forge
fftw                      3.3.10          nompi_h4024a70_101    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.1            h10f422b_1005    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
forcebalance              1.9.2            py39ha1f3e3e_0    conda-forge
freetype                  2.10.4               h4cff582_1    conda-forge
future                    0.18.2           py39h6e9494a_4    conda-forge
geometric                 0.9.7.2                    py_0    conda-forge
gettext                   0.19.8.1          hd1a6beb_1008    conda-forge
gflags                    2.2.2             hb1e8313_1004    conda-forge
glog                      0.5.0                h25b26a9_0    conda-forge
greenlet                  1.1.2            py39h9fcab8e_1    conda-forge
grpc-cpp                  1.40.0               h8fdf118_2    conda-forge
h11                       0.12.0             pyhd8ed1ab_0    conda-forge
h2                        4.1.0            py39h6e9494a_0    conda-forge
h5py                      3.4.0           nompi_py39hbc6cb89_101    conda-forge
hdf4                      4.2.15               hefd3b78_3    conda-forge
hdf5                      1.12.1          nompi_h2f0ef1a_101    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
httpcore                  0.13.7             pyhd8ed1ab_0    conda-forge
httpx                     0.20.0           py39h6e9494a_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       69.1                 he49afe7_0    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
importlib-metadata        4.8.2            py39h6e9494a_0    conda-forge
importlib_metadata        4.8.2                hd8ed1ab_0    conda-forge
importlib_resources       5.4.0              pyhd8ed1ab_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
ipykernel                 6.4.2            py39h71a6800_0    conda-forge
ipython                   7.29.0           py39h71a6800_1    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
jbig                      2.1               h0d85af4_2003    conda-forge
jedi                      0.18.0           py39h6e9494a_3    conda-forge
jinja2                    3.0.2              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   hbcb3906_0    conda-forge
jsonschema                4.2.1              pyhd8ed1ab_0    conda-forge
jupyter_client            7.0.6              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.1            py39h6e9494a_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
khronos-opencl-icd-loader 2021.06.30           h0d85af4_1    conda-forge
kiwisolver                1.3.2            py39hf018cea_1    conda-forge
kombu                     5.2.1            py39h6e9494a_0    conda-forge
krb5                      1.19.2               hcfbf3a7_3    conda-forge
lcms2                     2.12                 h577c468_0    conda-forge
lerc                      3.0                  he49afe7_0    conda-forge
libblas                   3.9.0           12_osx64_openblas    conda-forge
libbrotlicommon           1.0.9                h0d85af4_6    conda-forge
libbrotlidec              1.0.9                h0d85af4_6    conda-forge
libbrotlienc              1.0.9                h0d85af4_6    conda-forge
libcblas                  3.9.0           12_osx64_openblas    conda-forge
libcurl                   7.79.1               hf45b732_1    conda-forge
libcxx                    12.0.1               habf9029_0    conda-forge
libdeflate                1.8                  h0d85af4_0    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libevent                  2.1.10               h815e4d9_4    conda-forge
libffi                    3.4.2                he49afe7_4    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_23    conda-forge
libgfortran5              9.3.0               h6c81a4c_23    conda-forge
libglib                   2.70.0               hf1fb8c0_1    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
liblapack                 3.9.0           12_osx64_openblas    conda-forge
libnetcdf                 4.8.1           nompi_h6609ca0_101    conda-forge
libnghttp2                1.43.0               h6f36284_1    conda-forge
libopenblas               0.3.18          openmp_h3351f45_0    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libprotobuf               3.18.1               hcf210ce_0    conda-forge
libsodium                 1.0.18               hbcb3906_1    conda-forge
libssh2                   1.10.0               h52ee1ee_2    conda-forge
libthrift                 0.15.0               hab56fdc_1    conda-forge
libtiff                   4.3.0                hd146c10_2    conda-forge
libutf8proc               2.6.1                h35c211d_0    conda-forge
libwebp-base              1.2.1                h0d85af4_0    conda-forge
libxcb                    1.13              h35c211d_1003    conda-forge
libxml2                   2.9.12               h7e28ab6_1    conda-forge
libxslt                   1.1.33               h1acebb3_3    conda-forge
libzip                    1.8.0                h8b0c345_1    conda-forge
libzlib                   1.2.11            h9173be1_1013    conda-forge
llvm-openmp               12.0.1               hda6cdc1_1    conda-forge
lxml                      4.6.4            py39hf41e7f8_0    conda-forge
lz4-c                     1.9.3                he49afe7_1    conda-forge
markupsafe                2.0.1            py39h89e85a6_1    conda-forge
matplotlib-base           3.4.3            py39hb07454d_1    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mdtraj                    1.9.6            py39h996af62_1    conda-forge
mistune                   0.8.4           py39h89e85a6_1005    conda-forge
mmpbsa-py                 16.0                     pypi_0    pypi
mock                      4.0.3            py39h6e9494a_2    conda-forge
more-itertools            8.10.0             pyhd8ed1ab_0    conda-forge
msgpack-python            1.0.2            py39hf018cea_2    conda-forge
nbclient                  0.5.5              pyhd8ed1ab_0    conda-forge
nbconvert                 6.2.0            py39h6e9494a_0    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
nest-asyncio              1.5.1              pyhd8ed1ab_0    conda-forge
netcdf-fortran            4.5.3           nompi_h9ed14b0_106    conda-forge
networkx                  2.6.3              pyhd8ed1ab_1    conda-forge
nglview                   3.0.3              pyh8a188c0_0    conda-forge
notebook                  6.4.5              pyha770c72_0    conda-forge
numexpr                   2.7.3            py39h4d6be9b_1    conda-forge
numpy                     1.21.4           py39h7eed0ac_0    conda-forge
ocl_icd_wrapper_apple     1.0.0                hbcb3906_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openff-bespokefit         27rc4904+28.g739f680.dirty          pypi_0    pypi
openff-forcefields        2.0.0              pyh6c4a22f_0    conda-forge
openff-fragmenter-base    0.1.2              pyhd8ed1ab_0    conda-forge
openff-qcsubmit           0.2.4              pyhd8ed1ab_0    conda-forge
openff-toolkit-base       0.10.1             pyhd8ed1ab_0    conda-forge
openff-utilities          0.1.1              pyh6c4a22f_0    conda-forge
openjpeg                  2.4.0                h6e7aa92_1    conda-forge
openmm                    7.6.0           py39h8d72adf_0_khronos    conda-forge
openssl                   1.1.1l               h0d85af4_0    conda-forge
orc                       1.7.0                hf163092_0    conda-forge
packaging                 21.2               pyhd8ed1ab_1    conda-forge
packmol                   20.010               h508aa58_0    conda-forge
packmol-memgen            1.2.1rc0                 pypi_0    pypi
pandas                    1.3.4            py39h4d6be9b_1    conda-forge
pandoc                    2.16.1               h0d85af4_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parmed                    3.4.3            py39h9fcab8e_1    conda-forge
parquet-cpp               1.5.1                         1    conda-forge
parso                     0.8.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 he49afe7_0    conda-forge
pdb4amber                 20.1                     pypi_0    pypi
perl                      5.32.1          1_h0d85af4_perl5    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5           py39hde42818_1002    conda-forge
pillow                    8.3.2            py39he9bb72f_0    conda-forge
pint                      0.18               pyhd8ed1ab_0    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               hbcb3906_0    conda-forge
plotly                    5.3.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0            py39h6e9494a_2    conda-forge
prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.22             pyha770c72_0    conda-forge
prompt_toolkit            3.0.22               hd8ed1ab_0    conda-forge
psutil                    5.8.0            py39h89e85a6_2    conda-forge
pthread-stubs             0.4               hc929b4f_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
py-cpuinfo                8.0.0              pyhd8ed1ab_0    conda-forge
pyarrow                   6.0.0           py39h4d6536f_0_cpu    conda-forge
pycairo                   1.20.1           py39hbe14034_1    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydantic                  1.8.2            py39h89e85a6_2    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pymbar                    3.0.5            py39h329c335_2    conda-forge
pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyhd8ed1ab_1    conda-forge
pyrsistent                0.18.0           py39h89e85a6_0    conda-forge
pysocks                   1.7.1            py39h6e9494a_4    conda-forge
pytables                  3.6.1            py39hfd850c7_4    conda-forge
pytest                    6.2.5            py39h6e9494a_1    conda-forge
pytest-asyncio            0.16.0             pyhd8ed1ab_0    conda-forge
pytest-celery             0.0.0a1            pyh9f0ad1d_1    conda-forge
pytest-cov                3.0.0              pyhd8ed1ab_0    conda-forge
python                    3.9.7           h1248fe1_3_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytraj                    2.0.6                    pypi_0    pypi
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0              py39h89e85a6_3    conda-forge
pyzmq                     22.3.0           py39h7fec2f1_1    conda-forge
qcelemental               0.23.0             pyhd8ed1ab_0    conda-forge
qcengine                  0.20.1             pyhd8ed1ab_0    conda-forge
qcportal                  0.15.6             pyhd8ed1ab_0    conda-forge
rdkit                     2021.09.2        py39h88273a1_0    conda-forge
re2                       2021.09.01           he49afe7_0    conda-forge
readline                  8.1                  h05e3726_0    conda-forge
redis-py                  3.5.3              pyh9f0ad1d_0    conda-forge
redis-server              6.2.6                h05f672f_0    conda-forge
reportlab                 3.5.68           py39hf37cc50_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_0    conda-forge
requests-mock             1.9.3              pyhd8ed1ab_0    conda-forge
rfc3986                   1.5.0              pyhd8ed1ab_0    conda-forge
rich                      10.13.0          py39h6e9494a_0    conda-forge
sander                    16.0                     pypi_0    pypi
scipy                     1.7.2            py39h056f1c0_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                58.5.3           py39h6e9494a_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smirnoff99frosst          1.1.0              pyh44b312d_0    conda-forge
snappy                    1.1.8                hb1e8313_3    conda-forge
sniffio                   1.2.0            py39h6e9494a_2    conda-forge
sqlalchemy                1.4.26           py39h89e85a6_1    conda-forge
sqlite                    3.36.0               h23a322b_2    conda-forge
starlette                 0.14.2             pyhd8ed1ab_0    conda-forge
tenacity                  8.0.1              pyhd8ed1ab_0    conda-forge
terminado                 0.12.1           py39h6e9494a_1    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.11               h5dbffcc_1    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     1.2.2              pyhd8ed1ab_0    conda-forge
tornado                   6.1              py39h89e85a6_2    conda-forge
torsiondrive              1.1.0              pyhd8ed1ab_0    conda-forge
tqdm                      4.62.3             pyhd8ed1ab_0    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
typing-extensions         3.10.0.2             hd8ed1ab_0    conda-forge
typing_extensions         3.10.0.2           pyha770c72_0    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
urllib3                   1.26.7             pyhd8ed1ab_0    conda-forge
uvicorn                   0.15.0           py39h6e9494a_2    conda-forge
vine                      5.0.0              pyhd8ed1ab_1    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
widgetsnbextension        3.5.2            py39h6e9494a_0    conda-forge
xmltodict                 0.12.0                     py_0    conda-forge
xorg-kbproto              1.0.7             h35c211d_1002    conda-forge
xorg-libice               1.0.10               h0d85af4_0    conda-forge
xorg-libsm                1.2.3             h0d85af4_1000    conda-forge
xorg-libx11               1.7.2                h0d85af4_0    conda-forge
xorg-libxau               1.0.9                h35c211d_0    conda-forge
xorg-libxdmcp             1.1.3                h35c211d_0    conda-forge
xorg-libxext              1.3.4                h0d85af4_1    conda-forge
xorg-libxt                1.2.1                h0d85af4_2    conda-forge
xorg-xextproto            7.3.0             h35c211d_1002    conda-forge
xorg-xproto               7.0.31            h35c211d_1007    conda-forge
xtb                       6.4.1                h67747af_1    conda-forge
xtb-python                20.2             py39h89e85a6_3    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                haf1e3a3_0    conda-forge
zeromq                    4.3.4                he49afe7_1    conda-forge
zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11            h9173be1_1013    conda-forge
zstd                      1.5.0                h582d3a0_0    conda-forge
```

</details>

**Comment by SimonBoothroyd** [comment]
Thanks for reporting this @j-wags - this is a bug which occurs by no fragments being generated, and hence no ids are passed to the QC generation stage.

**Comment by SimonBoothroyd** [comment]
Looking over this further this should also raise an error early on - the `spec-file` argument should expect a factory schema, not a fitting schema AFAIK...

**Comment by SimonBoothroyd** [comment]
So to summarise - the issue post here is caused by providing the CLI with an invalid factory schema which got past the validation, which itself does not provide the custom SMARTS pattern used in the blog post, which in turn means no fragments are generated, and which means the QC generation stage fails.

This error should be resolved if using the schema generated on the line

```workflow.export_factory("workflow.json")```

in the blog post.

The validation logic is now fixed (#89), and we are adding more concise error messages in case this happens in the future (#90, #92).

Feel free to re-open this if issues still persist.

**Comment by j-wags** [comment]
Awesome. Thanks for the great response, @SimonBoothroyd. I'll follow up with @bergazin to make sure this unblocks their work!
